coding/bwt_coder.hpp

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include "coding/huffman.hpp"
+#include "coding/varint.hpp"
+
+#include "base/bwt.hpp"
+#include "base/move_to_front.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <vector>
+
+namespace coding
+{
+class BWTCoder
+{
+public:
+  struct Params
+  {
+    size_t m_blockSize = 32000;
```

Comment:
Why not to pass m_blockSize as a runtime param? What if blockSize will be computed algorithmically and won't be known at the compile time?